skimage/filters/thresholding.py

```diff
@@ -490,3 +490,111 @@ def find_local_maxima(hist):
             return bin_centers[upper_bound]
         elif bias == 'mid':
             return bin_centers[(threshold + upper_bound) // 2]
+
+
+def threshold_mean(image):
+    """Return threshold value based on the mean of grayscale values.
+
+    Parameters
+    ----------
+    image : (N, M) ndarray
+        Grayscale input image.
+
+    Returns
+    -------
+    threshold : float
+        Upper threshold value. All pixels with an intensity higher than
+        this value are assumed to be foreground.
+
+    References
+    ----------
+    .. [1] C. A. Glasbey, "An analysis of histogram-based thresholding
+        algorithms," CVGIP: Graphical Models and Image Processing,
+        vol. 55, pp. 532-537, 1993.
+
+    Examples
+    --------
+    >>> from skimage.data import camera
+    >>> image = camera()
+    >>> thresh = threshold_mean(image)
+    >>> binary = image > thresh
+    """
+    assert_nD(image, 2)
```

Comment:
We have a small piece of ambitions [here](http://scikit-image.org/docs/stable/user_guide/numpy_images.html#a-note-on-time).
Frankly speaking, 4D+ spatial data does not make sense to me at all. I suppose, the dream is to have 1timeD+3spatialD+1channelD for all image processing algorithms, but that is, of course, not really feasible. The main message is: if it is easy and cheap to add 3spatialD (or even NspatialD) support, one shouldn't hesitate to do that. :grey_exclamation: